pkg/config/config.go

```diff
@@ -43,3 +44,46 @@ func unmarshalConfig(filepath string, msg proto.Message) error {
 
 	return nil
 }
+
+func unmarshal(json string, msg proto.Message) error {
+	u := jsonpb.Unmarshaler{AllowUnknownFields: true}
+	return u.Unmarshal(strings.NewReader(json), msg)
+}
+
+// FromConfigMap builds a configuration by reading a map with the keys "global"
+// and "proxy", each containing JSON values.
+func FromConfigMap(configMap map[string]string) (*pb.All, error) {
```

Comment:
It doesn't look like this is being used here. Is it part of `upgrade`?

Reply:
yeah, i just merged this whole file from the upgrade branch...

Reply:
I'll remove for now and re-add as it's used.